### PR TITLE
Make go generate directives work on more setups.

### PR DESCRIPTION
**Pull request by shurcooL**
Similar to https://github.com/sourcegraph/go-sourcegraph/pull/49 and https://github.com/sourcegraph/srclib/pull/177.

Also document the go generate dependencies in README, largely based on [go-sourcegraph README development section](https://github.com/sourcegraph/go-sourcegraph/blob/nodb/README.md#development), with appropriate changes for this repo, and updates.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,8 +45,40 @@ The goal is to have all supported backends at feature parity, but until then, co
...
+===========
+
+### First-time installation of protobuf and other codegen tools
+
+You need to install and run the protobuf compiler before you can regenerate Go code after you change the `vcs.proto` file.
+
+1.	**Install protoc**, the protobuf compiler. Find more details in the [protobuf README](https://github.com/google/protobuf).
+
+	```
+	brew install --devel protobuf
```
Comment (sqs) [inline README.md]: Mention that this is for Mac only. E.g.:

> On Mac, run `brew install --devel protobuf`.

Comment (shurcooL) [inline README.md]: Should we apply the same change at https://github.com/sourcegraph/go-sourcegraph/blob/nodb/README.md#first-time-installation-of-protobuf-and-other-codegen-tools?

Comment (shurcooL) [inline README.md]: Also, minor, but I would say "on OS X" since that's the name of the operating system. Mac is the name of hardware.

Comment (sqs) [inline README.md]: Yes - and good point. Will change to OS X.

**Pull request merged**